pkg/controller/common/state.go

```diff
@@ -184,11 +212,16 @@ func (s *StateHandler) Update(reason string) error {
 		if s.ext.Status.State == nil {
 			s.ext.Status.State = &runtime.RawExtension{}
 		}
-		s.ext.Status.State.Raw, err = json.Marshal(wire)
+		data, err := json.Marshal(wire)
 		if err != nil {
 			s.Info("marshalling failed", "error", err)
 			return err
 		}
+		s.ext.Status.State.Raw, err = CompressEntriesState(data)
+		if err != nil {
+			s.Info("compressing failed", "error", err)
```

Comment:
Nit: What about using `s.Error()` instead?
```suggestion
			s.Error(err, "compressing failed")
```